marshal/consumer.go

```diff
@@ -375,6 +394,35 @@ func (c *Consumer) claimTopics() {
 		}
 	}
 
+	// let's compare the new topic claims vs old ones. Upon change, we should trigger an update
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	changed := false
+	// check for missing topic claims
+	for topic := range c.claimedTopics {
+		if !latestClaims[topic] {
+			changed = true
+			break
+		}
+	}
+	// check for new topic claims
+	if !changed && len(latestClaims) != len(c.claimedTopics) {
+		changed = true
+	}
+
+	if changed {
+		c.lock.RUnlock()
+		c.lock.Lock()
+		c.claimedTopics = latestClaims
+		c.lock.Unlock()
+		c.lock.RLock()
+		select {
+		case <-c.topicClaimsChan:
+		default:
+		}
+		c.topicClaimsChan <- c.claimedTopics
```

Comment:
It can never block because there is a single caller to managedClaims() and we always drain topicClaimChan before we write.